src/clr-angular/utils/smart-popover/index.ts

```diff
@@ -0,0 +1,18 @@
+/*
+ *  Copyright (c) 2016-2019 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ *
+ */
+
+export * from './enums/alignment.enum';
```

Comment:
Just a general question, are we planning to support all of these things are part of the public API? I thought we were planning to start off with it private, so we wouldn't necessarily have to export anything?

Reply:
I changed my mind on this. I'd like to make them part of the public API - that makes it simpler to handle close clicks inside the content container. Instead of a complex service+multiple event listeners to mimic how it was done before I would like to assume we ignore all clicks unless specifically declared in the popover template e.g. row actions was speced to close on click but filters and column toggle were not. 

If its not a good idea to expose everything I'd still like to expose the directives.

Reply:
I like the idea of the directives exposed to help with things like closing. I'm unsure about the other APIs since we would be locking into long term API support. Maybe we could mark it as an experimental public API to get feedback but not make long term support promises if something has to change?

Reply:
The plan is to expose it as ClrPopoverModuleNext like the way forms was handled.